### ISSUE TITLE
Allow using stdin

### ISSUE THREAD

**Issue by vasi** [martinellimarco/t2sz#8]
Currently t2sz mmaps the input file, so it has to be a physical file. This makes t2sz difficult to use in pipelines, where some stream is generated and needs to be compressed, but we don't want the uncompressed stream persisted to disk.

There's no particular reason we couldn't just read blocks into a buffer, though.

**Comment by martinellimarco** [comment]
Hey there,

You are right. At the moment, this is a design limitation that I should fix by replacing mmap with a buffered input, at least for raw mode. Tar mode requires some additional considerations.

I do plan to address this; however, at this time, I am quite busy with my daily work and can't promise any specific deadline.

It's a feature that I 100% intend to bring to the table, though.

**Comment by vasi** [comment]
That's reasonable, I know the feeling!

**Comment by csolisr** [comment]
Oh, that explains why I tried to do this:

```
t2sz -l22 -s1M -S10M -T1 -v -f -o example.tar.zst <(tar -cv ./example)
```

and it returned this:

```
ERROR: Unable to mmap '/dev/fd/63'
```

I would be using the "workaround" of creating the tar file manually and then compressing it with `t2sz`, but my free space in storage is currently smaller than the combined size of the currently uncompressed folder and its expected compressed size.

**Comment by martinellimarco** [comment]
Sorry @csolisr for this limitation. I haven’t had time to work on this project yet, but I plan to revisit it later this year.

**Comment by csolisr** [comment]
Understandable - fortunately it's not urgent for me to get this fixed, so take your time.

**Comment by vasi** [comment]
For now I've been using [zstdseek](https://github.com/SaveTheRbtz/zstd-seekable-format-go/tree/main/cmd/zstdseek) for this use case, but I look forward to getting this in t2z.

**Comment by martinellimarco** [comment]
Hi @vasi and @csolisr, sorry for the long delay.
Following up on the earlier request, stdin support is now available in v1.2.0.
Please let me know if you encounter any issues.

**Comment by csolisr** [comment]
Yeah, sorry for replying late, I've been busy as well. So, I fetch the latest version, compile it, and test it on the build folder itself, but the problem seems to remain, only slightly differently:

```
[csolisr@azkware-desktop build]$ t2sz -l22 -s1M -S10M -T1 -v -f -o build.tar.zst <(tar -cv .)
ERROR: Unable to seek '/dev/fd/63'
[csolisr@azkware-desktop build]$ ./
./build.tar.zst
./CMakeCache.txt
ls
build.tar.zst  CMakeCache.txt  CMakeFiles  cmake_install.cmake  CPackConfig.cmake  CPackSourceConfig.cmake  install_manifest.txt  Makefile  t2sz
[csolisr@azkware-desktop build]$ file build.tar.zst
build.tar.zst: empty
[csolisr@azkware-desktop build]$ 

```

Is it because I'm not supposed to create the original tarball that way?

**Comment by martinellimarco** [comment]
Thank you @csolisr,
It was not accounting for named pipes (like /dev/fd/63).
Now it has been updated to handle this and other cases.
